src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,48 @@
 
 class Animal {
   // write your code here
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
+Animal.alive = [];
+
+Animal.deleteDead = function(dead) {
+  for (let i = 0; i < Animal.alive.length; i++) {
+    if (Animal.alive[i].name === dead) {
+      Animal.alive.splice(i, 1);
+    }
+  }
+};
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(victim) {
+    if (victim.hasOwnProperty('hidden') && victim.hidden === false) {
```

Comment:
check if `herbivore instanceof Carnivore` as well
and do you need a `victim.hasOwnProperty('hidden')` check?